homeassistant/components/heos/config_flow.py

```diff
@@ -100,3 +116,72 @@ async def async_step_reconfigure(
             data_schema=vol.Schema({vol.Required(CONF_HOST, default=host): str}),
             errors=errors,
         )
+
+
+OPTIONS_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_USERNAME): selector.TextSelector(),
+        vol.Optional(CONF_PASSWORD): selector.TextSelector(
+            selector.TextSelectorConfig(type=selector.TextSelectorType.PASSWORD)
+        ),
+    }
+)
+
+
+class HeosOptionsFlowHandler(OptionsFlow):
+    """Define HEOS options flow."""
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Manage the options."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            authentication = CONF_USERNAME in user_input or CONF_PASSWORD in user_input
+            if authentication and CONF_USERNAME not in user_input:
+                errors[CONF_USERNAME] = "username_missing"
+            if authentication and CONF_PASSWORD not in user_input:
+                errors[CONF_PASSWORD] = "password_missing"
+
+            if not errors:
+                heos = cast(
+                    Heos, self.config_entry.runtime_data.controller_manager.controller
+                )
+
+                try:
+                    if authentication:
```

Comment:
if this is false this means nothing was entered, so we don't get here, no?

Reply:
We reach this because if nothing was entered `user_input` is an empty dictionary (vs. `None` before the user has submitted the screen). The tests confirm all of the logic branches are being reached as expected.

Reply:
right. I think I'd rather just do a `if not user_input` instead